embassy-sync/src/channel.rs

```diff
@@ -144,6 +158,20 @@ where
     pub fn try_recv(&self) -> Result<T, TryRecvError> {
         self.channel.try_recv()
     }
+
+    /// Allows a poll_fn to poll until the channel is ready to receive
+    ///
+    /// See [`Channel::poll_ready_to_receive()`]
+    pub fn poll_ready_to_receive(&self, cx: &mut Context<'_>) -> Poll<()> {
```

Comment:
we should be consistent on `recv` vs `receive`. Not sure which though. `receive` is nicer because no abbreviations, but `recv` would avoid a breaking change.

Reply:
If you want, we can move on to receive, and add a `#[deprecated]` note.

Reply:
Another argument for changing: we're breaking the API already by moving to `Poll<()>` over `bool`.

Reply:
I took the liberty of commiting my suggestion in 96a5888, I can easily drop/alter it again if we decide on something else.